plugin-testing-framework/script/test.ts

```diff
@@ -0,0 +1,470 @@
+export const shouldSetup: boolean = testUtil.readMochaCommandLineFlag(SETUP_OPTION_NAME);
+export const restartEmulators: boolean = testUtil.readMochaCommandLineFlag(RESTART_EMULATORS_OPTION_NAME);
+
+//// SERVER VARIABLES
+// Control how the server responds to update checks, test messages, and downloads.
+
+/** Response the server gives the next update check request */
+export var updateResponse: any;
+
+/** Response the server gives the next test message request */
+export var testMessageResponse: any;
+
+/** Called after the next test message request */
+export var testMessageCallback: (requestBody: any) => void;
+
+/** Called after the next update check request */
+export var updateCheckCallback: (requestBody: any) => void;
+
+/** Location of the update package given in the update check response */
+export var updatePackagePath: string;
+
+//////////////////////////////////////////////////////////////////////////////////////////
+// Use these classes to create and structure the tests
+
+export class TestBuilder {
+    public only: boolean;
+    public skip: boolean;
+    
+    constructor(options?: { only?: boolean, skip?: boolean }) {
+        if (!options) return;
+        
+        if (!!options.only) this.only = options.only;
+        if (!!options.skip) this.skip = options.skip;
+    }
+    
+    /**
+     * Called to create the test suite by the initializeTests function
+     * 
+     * coreTestsOnly - Whether or not only core tests are to be run
+     * projectManager - The projectManager instance that these tests are being run with
+     * targetPlatform - The platform that these tests are going to be run on
+     */
+    public create(coreTestsOnly: boolean, projectManager: tm.ProjectManager, targetPlatform: platform.IPlatform): void {};
+}
+
+/** Use this class to create a mocha.describe that contains additional tests */
+export class TestBuilderDescribe extends TestBuilder {
+    /** The name passed to the describe */
+    private describeName: string;
+    /** The path to the scenario that will be loaded by the test app for the nested TestBuilder objects */
+    private scenarioPath: string;
+    /** An array of nested TestBuilder objects that this describe contains */
+    private testBuilders: TestBuilder[];
+    /** Whether or not this.testBuilders directly contains any TestBuilderIt objects */
+    private hasIts: boolean;
+    /** Whether or not this.testBuilders directly contains any TestBuilder objects with an only flag */
+    private hasOnly: boolean;
+    
+    /**
+     * describeName - used as the description in the call to describe
+     * scenarioPath - if specified, will be set up before the tests run
+     * testBuilders - the testBuilders to create within this describe call
+     * only - if true, use describe.only
+     */
+    constructor(describeName: string, testBuilders: TestBuilder[], scenarioPath?: string, options?: { only?: boolean, skip?: boolean }) {
```

Comment:
Can we have `TestBuilder.describe(...)` and `TestBuilder.it(...)` wrapper functions that wrap around the construction of these new classes so that it mimics the mocha API as much as possible? That way contributors would still know how to work with our tests without having to learn too many new things.